x/leverage/keeper/iter_test.go

```diff
@@ -0,0 +1,119 @@
+package keeper_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (s *IntegrationTestSuite) TestGetEligibleLiquidationTargets_OneAddrOneAsset() {
+	app, ctx, require := s.app, s.ctx, s.Require()
+
+	// creates account which has supplied and collateralized 1000 uumee
+	addr := s.newAccount(coin(umeeDenom, 1000))
+	s.supply(addr, coin(umeeDenom, 1000))
+	s.collateralize(addr, coin("u/"+umeeDenom, 1000))
+
+	// user borrows 250 umee (max current allowed)
+	s.borrow(addr, coin(umeeDenom, 250))
+
+	zeroAddresses, err := app.LeverageKeeper.GetEligibleLiquidationTargets(ctx)
+	require.NoError(err)
+	require.Equal([]sdk.AccAddress{}, zeroAddresses)
+
+	// Note: Setting umee liquidation threshold to 0.05 to make the user eligible to liquidation
+	umeeToken := newToken("uumee", "UMEE")
+	umeeToken.CollateralWeight = sdk.MustNewDecFromStr("0.05")
+	umeeToken.LiquidationThreshold = sdk.MustNewDecFromStr("0.05")
+
+	require.NoError(app.LeverageKeeper.SetTokenSettings(ctx, umeeToken))
+
+	targetAddress, err := app.LeverageKeeper.GetEligibleLiquidationTargets(ctx)
+	require.NoError(err)
+	require.Equal([]sdk.AccAddress{addr}, targetAddress)
+}
+
+func (s *IntegrationTestSuite) TestGetEligibleLiquidationTargets_OneAddrTwoAsset() {
+	app, ctx, require := s.app, s.ctx, s.Require()
+
+	// creates account which has supplied and collateralized 1000 uumee
+	addr := s.newAccount(coin(umeeDenom, 1000))
+	s.supply(addr, coin(umeeDenom, 1000))
+	s.collateralize(addr, coin("u/"+umeeDenom, 1000))
+
+	// user borrows 250 umee (max current allowed)
+	s.borrow(addr, coin(umeeDenom, 250))
+
+	zeroAddresses, err := app.LeverageKeeper.GetEligibleLiquidationTargets(ctx)
+	require.NoError(err)
+	require.Equal([]sdk.AccAddress{}, zeroAddresses)
+
+	// mints and send to addr 100 atom and already
+	// enable 50 u/atom as collateral.
+	s.fundAccount(addr, coin(atomDenom, 100_000000))
+	s.supply(addr, coin(atomDenom, 50_000000))
+	s.collateralize(addr, coin("u/"+atomDenom, 50_000000))
+
+	// user borrows 4 atom (max current allowed - 1) user amount enabled as collateral * CollateralWeight
+	// = (50 * 0.1) - 1
+	// = 4app.
+	s.borrow(addr, coin(atomDenom, 4_000000))
+
+	// Note: Setting umee liquidation threshold to 0.05 to make the user eligible for liquidation
+	umeeToken := newToken("uumee", "UMEE")
+	umeeToken.CollateralWeight = sdk.MustNewDecFromStr("0.05")
+	umeeToken.LiquidationThreshold = sdk.MustNewDecFromStr("0.05")
```

Comment:
this 3 lines are repeated often. Could be better to create a function: `newUmeeToken(collateraW, liquidationT)`